### ISSUE TITLE
Version confusion surrounding: expected a string: "null"

### ISSUE THREAD

**Issue by steffeli** [semantic-release/commit-analyzer#117]
So after getting my Azure Pipeline up and running smoothly (sort of), I stumbled upon another problem with the commit-analyzer. I read https://github.com/semantic-release/commit-analyzer/issues/111 as the issue is identical. But for some reason I can't seem to get it working with either version: `7.0.0-beta.1` nor `7.0.0-beta-.2`. For some weird reason using a fresh `yarn init` and running `yarn add @semantic-release/commit-analyzer@beta` it installs the correct version, but the content in the node_modules is not what I expect. Looking at the source assets from the GitHub release this line: https://github.com/semantic-release/commit-analyzer/blob/31dadc7ba4293f63d38ac16c584d6de29c4eb54d/lib/analyze-commit.js#L26 is not present in my fresh install. The `isString(obj)` is still missing which results in `expected a string: "null"` for commits containing `type: null. I've tried to do a `yarn cache clean` to see if that does anything, but no dice.

After a fresh install adding the beta-version of commit-analyzer the file `analyze-commit.js`:
```
const {isMatchWith, isString} = require('lodash');
const micromatch = require('micromatch');
const debug = require('debug')('semantic-release:commit-analyzer');
const RELEASE_TYPES = require('./default-release-types');
const compareReleaseTypes = require('./compare-release-types');

/**
 * Find all the rules matching and return the highest release type of the matching rules.
 *
 * @param {Array} releaseRules the rules to match the commit against.
 * @param {Commit} commit a parsed commit.
 * @return {string} the highest release type of the matching rules or `undefined` if no rule match the commit.
 */
module.exports = (releaseRules, commit) => {
  let releaseType;

  releaseRules
    .filter(
      ({breaking, revert, release, ...rule}) =>
        // If the rule is not `breaking` or the commit doesn't have a breaking change note
        (!breaking || (commit.notes && commit.notes.length > 0)) &&
        // If the rule is not `revert` or the commit is not a revert
        (!revert || commit.revert) &&
        // Otherwise match the regular rules
        isMatchWith(commit, rule, (obj, src) => (isString(src) ? micromatch.isMatch(obj, src) : undefined))
    )
    .every(match => {
      if (compareReleaseTypes(releaseType, match.release)) {
        releaseType = match.release;
        debug('The rule %o match commit with release type %o', match, releaseType);
        if (releaseType === RELEASE_TYPES[0]) {
          debug('Release type %o is the highest possible. Stop analysis.', releaseType);
          return false;
        }
      } else {
        debug(
          'The rule %o match commit with release type %o but the higher release type %o has already been found for this commit',
          match,
          match.release,
          releaseType
        );
      }
      return true;
    });

  return releaseType;
};
```

**Comment by pvdlg** [comment]
Just install it with `yarn add @semantic-release/commit-analyzer@7.0.0-beta.1`

**Comment by steffeli** [comment]
I've tried that. The problem to me seems that the version published to NPM does not contain the latest changes which checks if the obj is also a string. From the file located at `node_modules/@semantic-release/commit-analyzer/lib/analyze-commit.js`, the code at the line 25:

`isMatchWith(commit, rule, (obj, src) => (isString(src) ? micromatch.isMatch(obj, src) : undefined))`

This is with a fresh install using `yarn init; yarn add @semantic-release/commit-analyzer@7.0.0-beta.1`.

Looking at the latest changes the line should be (I presume): 

`        isMatchWith(commit, rule, (obj, src) =>
          isString(src) && isString(obj) ? micromatch.isMatch(obj, src) : undefined
        )`

The problem I'm having is that `obj` becomes null as it doesn't contain the property `type: null`, for commits that should be ignored. And since the `isString(obj)` is missing the `micromatch.isMatch(obj, src)` crashes.

**Comment by pvdlg** [comment]
I released `v7.0.0-beta.2` that should solve the problem.
